### PR TITLE
[FLINK-5710] Add ProcTime() function to indicate StreamSQL.

### PR DESCRIPTION
**Pull request by haohui**
This is the commit we used internally -- There is no unit tests associated with this PR. It simply serves as a reference point for #3302.

### CODE REVIEW

**Comment by fhueske** [comment]
Thanks @haohui! The only thing missing would in fact be a test. IMO, extending `WindowAggregateTest` by copying the methods and adapting them to `proctime` would be sufficient. 

Let's see if #3302 is continued or not. If not, I'd like to add this PR.
Thanks, Fabian

**Comment by huawei-flink** [comment]
@haohui thanks for the contribution. I merged your code, I will push it later today.

**Comment by fhueske** [comment]
Thanks for the update @haohui! 
PR looks good. 
Will merge it later.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/stream/sql/WindowAggregateTest.scala [hunk 1]
```diff
@@ -94,6 +94,27 @@ class WindowAggregateTest extends TableTestBase {
     streamUtil.verifySql(sql, expected)
   }
 
+  @Test
+  def testProcessingTime() = {
+    val sql = "SELECT COUNT(*) FROM MyTable GROUP BY FLOOR(proc() TO HOUR)"
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/stream/sql/WindowAggregateTest.scala]: `proc()` -> `proctime()` ;-)

**Pull request closed without merging**